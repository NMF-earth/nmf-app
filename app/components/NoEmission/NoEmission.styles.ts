import { StyleSheet } from "react-native";

import { Layout } from "style";

export default StyleSheet.create({
  container: {
    ...Layout.containerWithPadding,
  },
  textView: {
    alignItems: "center",
    flex: 1,
  },
  paragraph: {
    textAlign: "center",
    paddingVertical: 4,
  },
  header: {
    paddingVertical: 14,
  },
  button: {
    marginVertical: 20,
  },
});
