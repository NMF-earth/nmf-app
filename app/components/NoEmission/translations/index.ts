import en from "./en";
import fr from "./fr";
import de from "./de";
import sv from "./sv";

interface TranslationKeys {
  NO_EMISSION_TITLE: string;
  NO_EMISSION_THANKS: string;
  NO_EMISSION_START_USING_APP: string;
  NO_EMISSION_ADD_FIRST_EMISSION: string;
}

export { en, fr, de, sv, TranslationKeys };
