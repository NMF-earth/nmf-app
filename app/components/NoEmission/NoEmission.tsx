import React from "react";
import { View, Image } from "react-native";

import { Text, Button } from "../";
import ImagesAssets from "../../constants/ImagesAssets";
import styles from "./NoEmission.style";
import { t } from "../../utils/translations";

interface Props {
  addEmission: () => void;
}

export default function NoEmission(props: Props) {
  return (
    <View style={styles.container}>
      <View style={styles.imageView}>
        <Image
          style={styles.image}
          resizeMode="contain"
          source={ImagesAssets.components.nmfStickerEarth}
        />
      </View>
      <View style={styles.textView}>
        <Text.H1 style={styles.header}>{t("NO_EMISSION_TITLE")}</Text.H1>
        <Text.Primary style={styles.paragraph}>
          {t("NO_EMISSION_THANKS")}
        </Text.Primary>
        <Text.Primary style={styles.paragraph}>
          {t("NO_EMISSION_START_USING_APP")}
        </Text.Primary>
        <Button.Primary
          fullWidth
          style={styles.button}
          onPress={props.addEmission}
          textType={"Primary"}
        >
          <Text.Primary bold white>
            {t("NO_EMISSION_ADD_FIRST_EMISSION")}
          </Text.Primary>
        </Button.Primary>
      </View>
    </View>
  );
}
