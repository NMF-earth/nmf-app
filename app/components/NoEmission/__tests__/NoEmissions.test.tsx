import React from "react";
import { create } from "react-test-renderer";

import NoEmission from "../";

jest.unmock("../");

it("renders correctly NoEmission", () => {
  const tree = create(<NoEmission />).toJSON();
  expect(tree).toMatchSnapshot();
});
