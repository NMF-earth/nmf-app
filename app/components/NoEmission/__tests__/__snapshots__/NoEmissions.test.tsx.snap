// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly NoEmission 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <StickersImage
      sticker="earth"
    />
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text.H1
        style={
          {
            "paddingVertical": 14,
          }
        }
      >
        Hi there ðŸ‘‹
      </Text.H1>
      <Text.Primary
        style={
          {
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        Thanks for joining our fight against climate change!
      </Text.Primary>
      <Text.Primary
        style={
          {
            "paddingVertical": 4,
            "textAlign": "center",
          }
        }
      >
        To start tracking your CO2 emissions, tap the button below
      </Text.Primary>
      <Button.Primary
        fullWidth={true}
        icon="add"
        onPress={[Function]}
        style={
          {
            "marginVertical": 20,
          }
        }
        text="Add my first emission"
      />
    </View>
  </View>
</RCTScrollView>
`;
